Source/Parallelization/WarpXRegrid.cpp

```diff
@@ -223,15 +276,61 @@ WarpX::RemakeLevel (int lev, Real time, const BoxArray& ba, const DistributionMa
             }
         }
 
-        if (costs[lev] != nullptr) {
-            costs[lev].reset(new MultiFab(costs[lev]->boxArray(), dm, 1, 0));
-            costs[lev]->setVal(0.0);
+        if (WarpX::load_balance_costs_update_algo == LoadBalanceCostsUpdateAlgo::Timers)
+        {
+            if (costs[lev] != nullptr)
+            {
+                costs[lev].reset(new MultiFab(costs[lev]->boxArray(), dm, 1, 0));
+                costs[lev]->setVal(0.0);
+            }
+        } else if (WarpX::load_balance_costs_update_algo == LoadBalanceCostsUpdateAlgo::Heuristic)
+        {
+            if (costs_heuristic[lev] != nullptr)
+            {
+                costs_heuristic[lev].reset(new amrex::Vector<Real>);
+                const int nboxes = Efield_fp[lev][0].get()->size();
+                costs_heuristic[lev]->resize(nboxes, 0.0); // Initializes to 0.0?
+            }
         }
 
         SetDistributionMap(lev, dm);
-    }
-    else
+
+    } else
     {
         amrex::Abort("RemakeLevel: to be implemented");
     }
 }
+
+void
+WarpX::ComputeCostsHeuristic ()
+{
+    for (int lev = 0; lev <= finest_level; ++lev)
+    {
+        long n_particles = 0;
```

Comment:
```suggestion
```
If you're not using this variable to accumulate the number of particles, I don't think it is useful anymore.

Reply:
Removed.